### DISCUSSION TITLE
We created full-text search enabled gallery

### DISCUSSION THREAD

**Discussion by polyrand** [excalidraw/excalidraw#2838] [Show and tell]
Hi! First I wanted to say I'm incredibly grateful to have Excalidraw. I mean, I really think it works like magic. So thank you to all the contributors that made it possible.

I first used Excalidraw to present my degree's final dissertation at my university. I studied medicine, but the work was about some text clustering algorithms, so I had to explain most things from zero. Not only that, but I thought it would take me days to do it, then I discovered Excalidraw and in one morning I could draw all the things I needed to present.

At that time, I thought it could be cool to have some kind of gallery with Excalidraw components, but I forgot about it. A few weeks ago I came back to the idea, only to discover you had already created the libraries' subdomain, which is awesome. But I still wanted to have some kind of search engine for the components without having to fill my personal library with many drawings. So, I asked my best friend, who is an architect (the buildings one, not a software architect) and we created [excalidraw.art](https://excalidraw.art/). Now there are a bit more than 100 logos, mostly software related, but we want to keep growing it with more drawings. Maybe we add drawings to create comics, design rooms, etc.

My friend has taken a lot of care in making the drawings as good as possible (grouping components, making things color-fillable, etc). That takes time to draw. There is also a server side full-text search engine powering the whole thing. You can search for logos not just by name, but also by description or tags. We decided to make some logos paid to compensate for it, but it's not like we are thinking of building any business around it. It's a one-time payment that gives you full access to every logo forever. If people end up buying it, we also wanted to donate part of it back to Excalidraw and the other open-source libraries powering the site.

I'm sharing this to know your thoughts before releasing it more publicly. The site may have some rough edges, and we are addressing a few problems with the names (one logo may give you the contents of another one, or throw an error because of typos in the names).

If someone ever needs access to all the logos for teaching (or something similar) but can't afford it, I could give away some keys. As I said, the last thing we want is feeling we are taking advantage of Excalidraw, we want to grow the ecosystem.

**TL;DR**

We created [excalidraw.art](https://excalidraw.art/). It has a growing gallery of Excalidraw drawings and provides full-text search for them so that you don't need to add them to your personal library. We wanted to:

1. Say thank you so much for Excalidraw!
2. Know your thoughts about it.

**Comment by vjeux** [comment]
@polyrand this is awesome! I know it's a lot to ask but I was wondering if you'd be open to bring some of the features you've implemented like text search to https://libraries.excalidraw.com/, I feel like as the library ecosystem is just starting, it may be good to avoid forking it and all build on-top of the same place.

**Reply by polyrand** [reply]
Hi! I don't know if I understood. You mean implementing something so that adding new libraries does not require forking the excalidraw-libraries repo? I'm happy to discuss it and help to implement something similar.

**Comment by polyrand** [comment]
I forgot to say that if you don't want us to use that domain name, I can look into changing to another one.

**Reply by dwelle** [reply]
Hey @polyrand,

Yes, we think the Excalidraw brand name should only be used within the Excalidraw organization. Otherwise, it will cause confusion, which is especially true when it comes to domain names.

**Reply by polyrand** [reply]
Hi @dwelle, thank you for the feedback. I didn't think too much about the domain name, and maybe I rushed into getting it. I'm already looking into getting another domain and change the name. Sorry for the inconvenience!

**Comment by polyrand** [comment]
Thanks everyone for your feedback!

I've already moved to another domain and I won't be using the excalidraw.art domain to show it. Also, I don't mean to create a fork / parallel / alternative library to the native one. Maybe I initially put too much emphasis on the integration with Excalidraw, but I wanted to make it more like a general set of icons/visuals, not only for Excalidraw, but for whatever people want (PowerPoint presentations, writing documents, teaching, etc.). I hope I didn't cause any issues for you, I'm very thankful to have Excalidraw ❤️.

If you have any other feedback, it's 100% welcome.

Apart from that, if you want to talk about implementing some kind of search for the native excalidraw libraries, I'm happy to help or discuss different ideas.

**Comment by lipis** [comment]
Awesome.. we merged it on our library.. feel free to update it there as well if you want to keep them in sync. As for a help.. we need to add this support to the core somehow.. to be able to show/edit elements in the library and then as a next step update our website.
